src/test/java/org/tarantool/TarantoolControl.java

```diff
@@ -216,44 +193,67 @@ public void waitStarted(String instanceName) {
     }
 
     /**
-     * Wait until the instance will be stopped.
+     * Sends <code>stop</code> command using tarantoolctl.
      *
-     * Use tarantoolctl status instanceName.
+     * @param instanceName target instance name
      */
-    public void waitStopped(String instanceName) {
-        while (status(instanceName) != 1) {
-            sleep();
+    public void stop(String instanceName) {
+        executeControlCommand("stop", instanceName);
+    }
+
+    /**
+     * Sends <code>stop</code> command using tarantoolctl
+     * and blocks until the process will be down.
+     *
+     * @param instanceName target instance name
+     */
+    public void stopAndAwait(String instanceName) {
+        ProcessCommand fetchInstancePidCommand =
+            new ProcessCommand(buildInstanceEnvironment(instanceName), "cat", instanceName + ".pid");
+        if (fetchInstancePidCommand.execute()) {
+            stop(instanceName);
+            waitStopped(fetchInstancePidCommand.getOutput(), instanceName);
         }
     }
 
-    public void start(String instanceName) {
-        executeCommand("start", instanceName);
+    /**
+     * Wait until the instance will be stopped.
+     *
+     * @param pid          instance PID
+     * @param instanceName target instance name
+     */
+    public void waitStopped(String pid, String instanceName) {
+        while (status(instanceName) != 1 || processAlive(pid)) {
+            sleep();
+        }
     }
 
-    public void stop(String instanceName) {
-        executeCommand("stop", instanceName);
+    private boolean processAlive(String pid) {
+        ProcessCommand findPsPidCommand =
+            new ProcessCommand(Collections.emptyMap(), "ps", "-o", "pid=", "-p", pid);
```

Comment:
Same thing here. Is there a native java class that do kill(pid, 0)?

Reply:
This one, unfortunately, cannot be run using Java API at least before 1.9 (see [ProcessHandle](https://docs.oracle.com/javase/9/docs/api/java/lang/ProcessHandle.html)).
There are two alternatives to do that `Runtime.exec(String)` or `ProcessBuilder.start()`. Both spawn subprocesses.

Reply:
So `kill -0 pid` is okay.